main/inc/lib/attendance.lib.php

```diff
@@ -2256,4 +2293,133 @@ public function exportAttendanceLogin($startDate, $endDate)
         $pdf = new PDF('A4', null, $params);
         $pdf->html_to_pdf_with_template($tableToString);
     }
+
+
+    /**
+     * Return all course for a student between dates and order by key date (Y-m-d)
+     * @param int           $student_id
+     * @param DateTime|null $startDate
+     * @param DateTime|null $endDate
+     * @param bool          $orderDesc
+     *
+     * @return array
+     */
+    public function getCoursesWithAttendance (
+        $student_id = 0,
+        DateTime $startDate = null,
+        DateTime $endDate = null,
+        $orderDesc = true
+    ) {
+
+        $courseManager = new CourseManager();
+        $attendanceLib = new Attendance();
+        $data = [];
+        // unused by now
+        /*
+         $specialCourses = $courseManager::returnSpecialCourses(
+            $student_id, false, false
+        );
+         */
+        $courses = $courseManager::returnCourses(
+            $student_id,
+            false,
+            false
+        );
+
+        /* Get course with (in_category) and without (not_category) category */
+        foreach ($courses as $coursesK => $coursesV) {
```

Comment:
$coursesK is not being use, no need to declare it.
You can just use $courseInfo or $courseItem or $courseData instead of $coursesV

Reply:
Currently, courses return two types of elements, in_category and not_category, these can be used later to distinguish a course. At the moment I will leave it as $coursesKey for possible use.

The special courses that would be processed in the same way have not been used either.